### PR TITLE
EHPR-94 | Setup remote db

### PR DESCRIPTION
**Pull request by fwpushan**
-----------

- [x]  Setup remote VPC + Security groups
- [x]  Setup rds cluster
- [x]  Inject env details for rds cluster to api
- [x]  Add jest capability to common
- [x]  Test infra code
Ticket: [EHPR-94](https://freshworks.atlassian.net/browse/EHPR-94)

### CODE REVIEW

File: terraform/db.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+resource "aws_db_subnet_group" "pgsql" {
+  name       = "pgsql"
+  subnet_ids = [aws_subnet.private_subnet.id]
+}
+
+resource "aws_rds_cluster_instance" "pgsql" {
+  count              = var.target_env == "prod" ? 2 : 1
+  identifier         = "${local.db_name}-${count.index}"
+  cluster_identifier = aws_rds_cluster.pgsql.id
+  instance_class     = "db.r5.large"
```
Comment (dbayly-freshworks) [inline terraform/db.tf:11]: This seems pretty big. How many records are we expecting?

**Review by dbayly-freshworks — APPROVED** [review]
Looks fine to me, one comment.

**Comment by hitankar-freshworks** [comment]
@fwpushan we will need a way to access the db locally. suggest we do what LTC has.

File: terraform/network.tf [hunk 2]
```diff
@@ -0,0 +1,56 @@
+resource "aws_vpc" "vpc" {
```
Comment (hitankar-freshworks) [inline terraform/network.tf:1]: Can we just not just go with a security group and since this is just temporary

File: terraform/db.tf (hunk 1 above)
Comment (fwpushan) [reply terraform/db.tf:11]: This code is copied from ltx and ets ..... Almost or a little bigger than HCAP.

File: terraform/network.tf (hunk 2 above)
Comment (fwpushan) [reply terraform/network.tf:1]: As per @v-y-a-s suggestion. We deployed it our sandboxed env and this will be destroyed.

**Comment by fwpushan** [comment]
> @fwpushan we will need a way to access the db locally. suggest we do what LTC has.

@hitankar-freshworks 
This is the first part of the sub-task and this will unblock other. 
In the next part I will setup bastion host.

File: terraform/db.tf (hunk 1 above)
Comment (vyasworks) [reply terraform/db.tf:11]: Maybe something smaller would do? I highly doubt we need R5 with 16 gb memory. 

Maybe use a general purpose category? https://aws.amazon.com/rds/instance-types/

File: terraform/network.tf (hunk 2 above)
Comment (vyasworks) [reply terraform/network.tf:1]: This might take a bit of time to setup but nevertheless, a good exercise. 

We will be using LZ2's VPC when we get access

**Comment by arranfw** [comment]
> > @fwpushan we will need a way to access the db locally. suggest we do what LTC has.
> 
> @hitankar-freshworks This is the first part of the sub-task and this will unblock other. In the next part I will setup bastion host.

Could you document the tasks in the ticket? https://freshworks.atlassian.net/browse/EHPR-29

File: packages/common/src/dto/skill-information.dto.spec.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+describe('Test Skill-Information DTO', () => {
```
Comment (vyasworks) [inline packages/common/src/dto/skill-information.dto.spec.ts:1]: Not sure if this belongs in this PR, but no issues :smile:

File: terraform/db.tf [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  engine             = aws_rds_cluster.pgsql.engine
+  engine_version     = aws_rds_cluster.pgsql.engine_version
+}
+
+resource "aws_rds_cluster" "pgsql" {
+  cluster_identifier  = local.db_name
+  engine              = "aurora-postgresql"
```
Comment (vyasworks) [inline terraform/db.tf:18]: I wonder what the difference in postgres and aurora postgres engine would be? 
Docs from aws say just postgres is also an option, tf docs says aurora.. Unsure about this. 

would be good to know the difference. 

https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-cluster.html

**Pull request merged**